src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -42,30 +60,35 @@ internal Tensor(IntPtr handle)
             public override bool Equals(object? obj)
             {
                 return (obj is Tensor) && this.Equals((obj as Tensor)!);
-
             }
 
             /// <summary>
             ///  TBD
             /// </summary>
             /// <returns></returns>
-            public override int GetHashCode()
-            {
-                return base.GetHashCode();
-            }
+            public override int GetHashCode() => base.GetHashCode();
 
             /// <summary>
             /// A friendly name for the tensor. This is useful for debugging purposes.
             /// </summary>
             public string? name { get; set; }
 
+            /// <summary>
+            /// Helper method to determine if a tensor is disposed or not.
+            /// </summary>
+            public bool IsDisposed => handle == IntPtr.Zero;
+
             /// <summary>
             ///   Finalize the tensor. Releases the tensor and its associated data.
             /// </summary>
             ~Tensor() => Dispose(false);
 
             public void Dispose()
             {
+                foreach (var callback in OnTensorDisposeCallback) {
```

Comment:
I'm skeptical about this. I would like to better understand the performance impact of having this call on every tensor dispose.

Reply:
There are the same number of calls when tensors are created, and deleting from a hash set is O(1), so I'm not sure it would be much of a problem?

Reply:
Yeah, as long as it's a constant-time operation, it's fine. The cost of removing items from a collection is always worrisome to me, and this is going to be on a lot of critical code paths, so I want to make sure we have it under control.

Reply:
Let's keep an eye on performance and revisit this if we see any degradation?

Reply:
I'll run some benchmarks before releasing this to NuGet.